spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaBeanPostProcessor.java

```diff
@@ -0,0 +1,234 @@
+                if (InjectionMetadata.needsRefresh(metadata, clazz)) {
+                    if (metadata != null) {
+                        metadata.clear(pvs);
+                    }
+                    metadata = buildLocalArmeriaPortMetadata(clazz);
+                    injectionMetadataCache.put(cacheKey, metadata);
+                }
+            }
+        }
+        return metadata;
+    }
+
+    private InjectionMetadata buildLocalArmeriaPortMetadata(Class<?> clazz) {
+        final List<InjectedElement> elements = new ArrayList<>();
+        Class<?> targetClass = clazz;
+
+        do {
+            final List<InjectionMetadata.InjectedElement> currElements = new ArrayList<>();
+
+            ReflectionUtils.doWithLocalFields(targetClass, field -> {
+                if (field.isAnnotationPresent(LocalArmeriaPort.class)) {
+                    if (Modifier.isStatic(field.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPort annotation is not supported on the static field: " +
+                                field.getName());
+                    }
+                    currElements.add(new LocalArmeriaPortElement(field, field, null));
+                } else if (field.isAnnotationPresent(LocalArmeriaPorts.class)) {
+                    if (Modifier.isStatic(field.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPorts annotation is not supported on the static field: " +
+                                field.getName());
+                    }
+                    currElements.add(new LocalArmeriaPortsElement(field, null));
+                }
+            });
+
+            ReflectionUtils.doWithLocalMethods(targetClass, method -> {
+                final Method bridgedMethod = BridgeMethodResolver.findBridgedMethod(method);
+                if (!BridgeMethodResolver.isVisibilityBridgeMethodPair(method, bridgedMethod)) {
+                    return;
+                }
+                if (bridgedMethod.isAnnotationPresent(LocalArmeriaPort.class)) {
+                    if (Modifier.isStatic(method.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPort annotation is not supported on the static field: " +
+                                method.getName());
+                    }
+                    final PropertyDescriptor pd = BeanUtils.findPropertyForMethod(bridgedMethod, clazz);
+                    currElements.add(new LocalArmeriaPortElement(method, bridgedMethod, pd));
+                } else if (bridgedMethod.isAnnotationPresent(LocalArmeriaPorts.class)) {
+                    if (Modifier.isStatic(method.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPorts annotation is not supported on the static field: " +
```

Comment:
on the static field -> on a static method

Reply:
Oops... it's my mistake.

Reply:
Fixed 44dfcbd